dbt/adapters/snowflake/connections.py

```diff
@@ -28,6 +28,10 @@
 _TOKEN_REQUEST_URL = 'https://{}.snowflakecomputing.com/oauth/token-request'
 
 
+@dataclass
+class SnowflakeAdapterResponse(AdapterResponse):
+    query_id: Optional[str] = None
```

Comment:
"Oh pedantry of types! you are my curse." ~~ Shakespeare or something

Concerning [the definition of the `AdapterResponse` class](https://github.com/dbt-labs/dbt-core/blob/434f3d26789d4bed3f0057c7c7d898e9dbf6687f/core/dbt/contracts/connection.py#L30), notice the `_msg` field is just a string and not an `Optional`. Why? Because there should always be a message string. And indeed, there's one even for bugged queries:

![image](https://user-images.githubusercontent.com/67295367/152120264-ae9307c2-5dcf-4767-a7a9-6b787cbe86e5.png)

Can a query_id be null? I'm expecting the answer to be "no" ([stored procedures](https://community.snowflake.com/s/article/Why-does-Snowflake-LAST-QUERY-ID-function-returns-NULL-when-I-try-to-get-the-query-ID-which-is-executed-in-stored-procedure) notwithstanding). If so, in the spirit of being good abstracters, let's remove the Optional :) Per this [stackoverflow thread](https://stackoverflow.com/questions/63892939/is-query-id-unique-among-all-the-snowflake-accounts?rq=1), looks like this field is even UUID-unique across all accounts -- cool!

Reply:
If we change the type here, change the test predicate to a plain `assert` statement instead of having the `is not` predicate (i.e. make sure a query_id is actually present!).